ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2/Forker.hs

```diff
@@ -165,6 +165,17 @@ implForkerCommit env = do
                   _ AS.:< closeOld' -> closeLedgerSeq (LedgerSeq closeOld')
                 -- Finally, close the anchor of @lseq@ (which is a duplicate of
                 -- the head of @olddb'@).
+                --
+                -- Note if the resource registry used to create the Forker is
+                -- ephemeral as it should (one created on each Chain selection
+                -- or each Forging loop iteration), this first duplicated state
+                -- will be closed by the resource registry closing down, so this
+                -- will be a double release, which is fine. We prefer keeping
+                -- this action just in case some client doesn't respect the
+                -- condition of registries being ephemeral.
```

Comment:
Hmm, do we document the requirement that one should use a `Forker` with a short-lived `ResourceRegistry` somewhere? This is especially weird as eg the LedgerDB state machine test uses a long-lived registry for all forkers.

Alternatively, I think this comment could be rephrased to say that we close this just to release resources as early as possible, even if it might be closed soon after due to the registry getting closed.

Reply:
I have reworded it a bit